stackslib/src/chainstate/nakamoto/miner.rs

```diff
@@ -76,6 +76,8 @@ use crate::net::Error as net_error;
 
 /// New tenure information
 pub struct NakamotoTenureStart {
+    /// tenure change transaction
+    pub tenure_change_tx: StacksTransaction,
```

Comment:
You may want to take a look at what I have in the tenure validation PR

Reply:
Yes, I'm familiar with the changes, and I expect to have to resolve conflicts on merge/rebase. It should be pretty easy to resolve. In a subsequent PR, I'd prefer to refactor what's happening with `chainstate::nakamoto::miner` -- I believe that the miner's instantiation routines (i.e., the `from_*`s) should be substantially refactored. Most of that tenure information is not actually important to the BlockBuilder struct-- really, it's just about supplying initial transactions to the builder (which the builder doesn't assemble, they're passed in).